math.go

```diff
@@ -8,11 +8,31 @@ type Point struct {
 	X, Y float32
 }
 
+type Line struct {
```

Comment:
We're trying to make sure that all code that gets put into master now is documented. Could you do that here?